t/comp/use.t

```diff
@@ -157,6 +157,12 @@ like $@, qr/^Can't use string/,
     like $warnings, qr/^Downgrading a use VERSION declaration to below v5.11 is deprecated, and will become fatal in Perl 5.40 at /,
         'use 5.01 after use 5.012 provokes deprecation warning';
 }
+eval 'use 5.039; use 5.020;';
+like $@, qr/^Downgrading a use VERSION declaration when above v5\.39 is not permitted at /,
+    'downgrading prevailing version in same scope is not permitted';
+eval '{ use 5.039; } use 5.020;';
+ok !$@, 'downgrading prevailing version by falling out of scope is permitted';
+
```

Comment:
shouldn't this test also availability of symbols ?

Reply:
also it would be nice to test behaviour when code already has exported symbol, eg:
```
use Scalar::Util qw (weaken);
use v5.40;
```
(and vice versa)

Reply:
Symbol visibility is tested by a different test; see the one on line 181.

Reply:
can you move them closer?

also related test:
```
eval "{ use v5.39 } use v5.20; true";
ok $@, ...
```